### ISSUE TITLE
independent padding

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#1203]
Feel like I already commented on this somewhere, but I can't find it.

It should be possible to set the `xAxisPadding` and `yAxisPadding` independently for each side. Often, for example, one wants to pad the top or right but not the bottom or left.

For a backward-compatible interface, we could do something like [what CSS does](https://developer.mozilla.org/en-US/docs/Web/CSS/padding) and if there's one value, apply it to all sides, if there are two values, then use them for right/left and top/bottom.

Possible interfaces:

```js
[right, left]
"right left"
{right: right, left: left}
```

I guess I prefer the last, because it avoids the ordering question. (CSS padding uses "top right bottom left" and applies top or right to bottom or left if missing.) 

Although it's possible to implement more than one interface, I'd prefer not to complicate it too much. I'm open to arguments for `[left, right]` etc.

**Comment by juanvalag2019** [comment]
Yes, this would be a good feature, I'm struggling with a barchart and the label in the highest bar is hidden.

**Comment by gordonwoodhull** [comment]
@juanvalag2019, adjusting the [margins](http://dc-js.github.io/dc.js/docs/html/MarginMixin.html#margins) should take care of most label issues. 

The feature in this issue would help with more obscure, specialized cases.
